widgets/linechart/linechart_test.go

```diff
@@ -960,6 +960,53 @@ func TestLineChartDraws(t *testing.T) {
 				return ft
 			},
 		},
+		{
+			desc:   "more values than capacity, X rescales with NaN values ignored",
```

Comment:
There are a couple more interesting cases we might want to test:

- all the values are NaN.
- the very first visible value is NaN.
- the very last visible value is NaN.

I would appreciate if you would consider adding the test cases. However I know this is a lot of work, so if you want you can just confirm that the corner cases work and I will be happy to add the missing test coverage after we merge your PR in.

If you do choose to add the tests, we could consider adding them cases with less values in the series (just the smallest amount we need for the test case). That should make them easier to write.

Reply:
You are right, no problem! I'll add them.